### PR TITLE
Header anchor

### PR DESCRIPTION
**Pull request by Akarys42**
This PR adds anchors to all HTML headers, to make creating anchored links easier.

For the link symbol to only show up on hover, QuiltMC/quilt-bulma#12 needs to be merged.

---
See preview on Cloudflare Pages: https://preview-122.quiltmc-org.pages.dev

### CODE REVIEW

File: astro.config.ts [hunk 1]
```diff
@@ -4,13 +4,17 @@ import astroI18next from "astro-i18next";
...
+import { fromHtml } from 'hast-util-from-html'
+
 
 // https://astro.build/config
 export default defineConfig({
 	site: "https://quiltmc.org",
 	integrations: [mdx(), astroI18next(), sitemap(), compress(), purgecss()],
 	markdown: {
 		syntaxHighlight: "prism",
+		rehypePlugins: [[rehypeAutolinkHeadings, { content: fromHtml('<span class="header-anchor-container icon pl-3"><i class="header-anchor fas fa-lg fa-link has-text-grey-dark is-size-5"></i></span>'), behavior: "append" }]],
```
Comment (pluiedev) [inline astro.config.ts:17]: `grey-dark` is *way* too dark for this — in fact, [it fails the WCAG color contrast standard](https://webaim.org/resources/contrastchecker/?fcolor=4A4A4A&bcolor=271F33) by quite a wide margin and we have to choose a brighter color for accessibility's sake

**Review by pluiedev — CHANGES_REQUESTED** [review]

File: astro.config.ts (hunk 1 above)
Comment (gdude2002) [reply astro.config.ts:17]: I would probably use the `link` colour.

**Review by pluiedev — APPROVED** [review]

**Comment by Akarys42** [comment]
Running `pnpm i` on multiple computers shows that no change is required, if you have any insights as to why the CI runs are failing, please do say so.

**Comment by pluiedev** [comment]
> Running `pnpm i` on multiple computers shows that no change is required, if you have any insights as to why the CI runs are failing, please do say so.

Did you upgrade to pnpm 8.0? It looks like the `lockfileVersion` field of the lock file has been bumped from 5.4 to 6.0, and there are major changes in the file format. Running `git diff --stat` after a fresh `pnpm i` yields this:
```
pnpm-lock.yaml | 1377 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------------------------------------------------------------
 1 file changed, 698 insertions(+), 679 deletions(-)
 ```

**Pull request merged**